async_simple/util/ThreadPool.h

```diff
@@ -93,19 +98,28 @@ inline ThreadPool::ThreadPool(size_t threadNum, bool enableWorkSteal)
 
             // If _enableWorkSteal false or work steal failed, wait for a pop
             // task.
-            if (!workerItem.fn && !_queues[id].pop(workerItem)) {
-                break;
-            }
+            if (!workerItem.fn && !_queues[id].pop(workerItem))
+                continue;
 
-            if (workerItem.fn) {
+            if (workerItem.fn)
                 workerItem.fn();
-            }
         }
     };
 
     _threads.reserve(_threadNum);
     for (auto i = 0; i < _threadNum; ++i) {
         _threads.emplace_back(worker, i);
+
+#ifdef __linux__
+        // Run threads per core.
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(i, &cpuset);
```

Comment:
1、cpu id maybe not start at `0` or be sequenced in container environment
2、`_threadNum` maybe great than available cpus
3、macOS support pthread？

Reply:
1. According to https://linux.die.net/man/3/cpu_set, the cpu id starts at 0 and is continuous.
2. I refactored it to `CPU_SET(i % CPU_COUNT(i), &cpuset);` So it wouldn't overflow.
3. I didn't find related things in https://developer.apple.com/search/?q=CPU_SET. Given the users who would need performance in macOS should be less, I think we could give up to  support them.